ctk/common/src/main/kotlin/org/codice/compliance/verification/core/SignatureAndEncryptionVerifier.kt

```diff
@@ -0,0 +1,85 @@
+ */
+package org.codice.compliance.verification.core
+
+import org.apache.cxf.rs.security.saml.sso.SSOConstants
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLCore_5_4_2_b
+import org.codice.compliance.SAMLCore_5_4_2_b1
+import org.codice.compliance.SAMLCore_6_1_b
+import org.codice.compliance.children
+import org.codice.compliance.recursiveChildren
+import org.codice.compliance.utils.TestCommon
+import org.w3c.dom.Node
+
+class SignatureAndEncryptionVerifier(private val node: Node) {
+    companion object {
+        private const val ENCRYPTED_DATA = "EncryptedData"
+    }
+
+    /**
+     * 5 SAML and XML Signature Syntax and Processing
+     * 6 SAML and XML Encryption Syntax and Processing
+     */
+    fun verify() {
+        // Signing
+        verifySignatureSyntaxAndProcessing()
+
+        // Encryption
+        verifyGeneralConsiderations()
+    }
+
+    /** 5.4.2 References */
+    private fun verifySignatureSyntaxAndProcessing() {
+        node.children(SSOConstants.SIGNATURE).forEach {
+            val references = it.recursiveChildren("Reference")
+            if (references.size != 1)
+                throw SAMLComplianceException.create(SAMLCore_5_4_2_b1,
+                        message = "A signature needs to have exactly one Reference, " +
+                                "${references.size} found.",
+                        node = node)
+
+            val uriValue = references[0].attributes.getNamedItem("URI")?.textContent
+                    ?: throw SAMLComplianceException.create(SAMLCore_5_4_2_b1,
+                            message = "URI attribute not found.",
+                            node = node)
+
+            val formattedId = "#${it.parentNode.attributes?.getNamedItem("ID")?.textContent}"
+            if (uriValue != formattedId)
+                throw SAMLComplianceException.createWithPropertyMessage(SAMLCore_5_4_2_b,
+                        property = "URI",
+                        actual = uriValue,
+                        expected = formattedId,
+                        node = node)
+        }
+    }
+
+    /** 6.1 - General Considerations **/
+    private fun verifyGeneralConsiderations() {
+        val elements = node.recursiveChildren("EncryptedAssertion") +
+                node.recursiveChildren("EncryptedID") +
+                node.recursiveChildren("EncryptedAttribute")
+
+        elements.flatMap { it.children("EncryptedData") }
+                .forEach {
+                    val type = it.attributes.getNamedItem("Type").textContent
+                    if (type != TestCommon.ELEMENT)
+                        throw SAMLComplianceException.createWithPropertyMessage(SAMLCore_6_1_b,
+                                property = "$ENCRYPTED_DATA's Type attribute",
+                                actual = type,
+                                expected = TestCommon.ELEMENT,
+                                node = node)
+                }
```

Comment:
❗️ This whole function is already being covered by @bakejeyner's decrypt PR. Could you just add the `SAMLCore_6_1_b` message onto the existing messages in the Exception he throws in there? https://github.com/connexta/saml-conformance/blob/master/ctk/common/src/main/kotlin/org/codice/compliance/verification/core/EncryptionVerifier.kt#L70